syntax = "proto3";

option go_package = "github.com/milkyhoop/flow-executor/internal/proto/tenant_manager";
package tenant_manager;

import "google/protobuf/empty.proto";
import "google/protobuf/timestamp.proto";

service TenantManager {
  rpc CreateTenant (CreateTenantRequest) returns (TenantResponse);
  rpc GetTenant (GetTenantRequest) returns (TenantResponse);
  rpc UpdateTenant (UpdateTenantRequest) returns (TenantResponse);
  rpc DeleteTenant (DeleteTenantRequest) returns (google.protobuf.Empty);
  rpc ListTenants (google.protobuf.Empty) returns (ListTenantsResponse);

  rpc HealthCheck (google.protobuf.Empty) returns (google.protobuf.Empty);
}

message Tenant {
  string id = 1;
  string alias = 2;
  string display_name = 3;
  string menu_items = 4; // JSON string
  string address = 5;
  string status = 6;
  google.protobuf.Timestamp created_at = 7;
  google.protobuf.Timestamp updated_at = 8;
}

message CreateTenantRequest {
  string alias = 1;
  string display_name = 2;
  string menu_items = 3; // JSON string
  string address = 4;
}

message GetTenantRequest {
  string alias = 1;
}

message UpdateTenantRequest {
  string alias = 1;
  string display_name = 2;
  string menu_items = 3; // JSON string
  string address = 4;
  string status = 5;
}

message DeleteTenantRequest {
  string alias = 1;
}

message TenantResponse {
  string status = 1;
  Tenant tenant = 2;
}

message ListTenantsResponse {
  repeated Tenant tenants = 1;
}
